File: typescript/2022/day1/main.ts
import * as fs from "fs";

// Part 1
console.log(
  Math.max(
    ...fs
      .readFileSync("typescript/2022/day1/test.txt", "utf-8")
      .split("\r\n\r\n")
      .map((elf: string) => {
        return elf.split("\r\n").reduce((total, current) => {
          return total + Number(current);
        }, 0);
      })
  )
);

// Part 2
console.log(
  fs
    .readFileSync("typescript/2022/day1/test.txt", "utf-8")
    .split("\r\n\r\n")
    .map((elf: string) => {
      return elf.split("\r\n").reduce((total, current) => {
        return total + Number(current);
      }, 0);
    })
    .sort((a: number, b: number) => b - a)
    .slice(0, 3)
    .reduce((total: number, current: number) => {
      return total + current;
    })
);

File: python/2022/day1/main.py
# Part 1
print(max([sum([int(i) for i in x.split("\n")])for x in open("python/2022/day1/test.txt").read().split("\n\n")]))
# Part 2
print(sum(sorted([sum([int(i) for i in x.split("\n")])for x in open("python/2022/day1/test.txt").read().split("\n\n")])[-3:]))